python/erdos/streams.py

```diff
@@ -45,8 +51,8 @@ class Stream(ABC):
         This class should never be initialized manually.
     """
 
-    def __init__(self, internal_stream: PyStream):
-        self._internal_stream = internal_stream
+    def __init__(self, internal_stream: PyStream) -> None:
+        self._internal_stream: Stream[T] = internal_stream
```

Comment:
```suggestion
        self._internal_stream: PyStream = internal_stream
```
The `internal_stream` carries the same type that it is being assigned from.